src/library/kernels/SpGEMM_computeNnzCt_kernels.cl

```diff
@@ -0,0 +1,70 @@
+R"(
+/* ************************************************************************
+* The MIT License (MIT)
+* Copyright 2014-2015 weifengliu
+*  Permission is hereby granted, free of charge, to any person obtaining a copy
+*  of this software and associated documentation files (the "Software"), to deal
+*  in the Software without restriction, including without limitation the rights
+*  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+*  copies of the Software, and to permit persons to whom the Software is
+*  furnished to do so, subject to the following conditions:
+ 
+*  The above copyright notice and this permission notice shall be included in
+*  all copies or substantial portions of the Software.
+ 
+*  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+*  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+*  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+*  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+*  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+*  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+*  THE SOFTWARE.
+* ************************************************************************ */
+/* ************************************************************************
+*  < A CUDA/OpenCL General Sparse Matrix-Matrix Multiplication Program >
+*  
+*  < See papers:
+*  1. Weifeng Liu and Brian Vinter, "A Framework for General Sparse 
+*      Matrix-Matrix Multiplication on GPUs and Heterogeneous 
+*      Processors," Journal of Parallel and Distributed Computing, 2015.
+*  2. Weifeng Liu and Brian Vinter, "An Efficient GPU General Sparse
+*      Matrix-Matrix Multiplication for Irregular Data," Parallel and
+*      Distributed Processing Symposium, 2014 IEEE 28th International
+*      (IPDPS '14), pp.370-381, 19-23 May 2014.
+*  for details. >
+* ************************************************************************ */ 
+
+#define GROUPSIZE_256 256
+
```

Comment:
This looks similar to spmv algorithm where we are traversing CSR matrix. Maybe it might be accelerated using vectorized version. We did that in csr2dense.